### PR TITLE
Update outdated links in `PULL_REQUEST_TEMPLATE.md`

### PR DESCRIPTION
**Pull request by dklimpel**
Fixes: #11214

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,10 +3,11 @@
 <!-- Please read CONTRIBUTING.md before submitting your pull request -->
```
Comment (DMRobertson) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: Can we also change this comment to refer to `https://matrix-org.github.io/synapse/latest/development/contributing_guide.html`)?

File: changelog.d/11225.misc [hunk 2]
```diff
@@ -0,0 +1 @@
+Update outdated links in contribution comments (`PULL_REQUEST_TEMPLATE.md`).
```
Comment (DMRobertson) [inline changelog.d/11225.misc:1]:
```suggestion
Replace outdated links in the pull request checklist with links to the rendered documentation.
```

**Review by DMRobertson — CHANGES_REQUESTED** [review]
Thanks, we must have missed this with a recent move towards using the rendered doc. A few small requests:

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (dklimpel) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
<!-- Please read https://matrix-org.github.io/synapse/latest/development/contributing_guide.html before submitting your pull request -->
```

**Review by DMRobertson — APPROVED** [review]
Brilliant. Thank you for spotting and fixing this!

**Pull request merged**